lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/service/ReportUserIdentityService.java

```diff
@@ -0,0 +1,129 @@
+
+public class ReportUserIdentityService {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private final Gpg45ProfileEvaluator gpg45ProfileEvaluator;
+
+    public ReportUserIdentityService(Gpg45ProfileEvaluator gpg45ProfileEvaluator) {
+        this.gpg45ProfileEvaluator = gpg45ProfileEvaluator;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public ReportUserIdentityService() {
+        this.gpg45ProfileEvaluator = new Gpg45ProfileEvaluator();
+    }
+
+    public Optional<Vot> getStrongestAttainedVotForCredentials(List<VerifiableCredential> vcs)
+            throws ParseException {
+        for (Vot votToCheck : SUPPORTED_VOTS_BY_DESCENDING_STRENGTH) {
+            boolean requestedVotAttained;
+            if (votToCheck.getProfileType().equals(GPG45)) {
+                requestedVotAttained =
+                        achievedWithGpg45Profile(
+                                votToCheck, VcHelper.filterVCBasedOnProfileType(vcs, GPG45));
+            } else {
+                requestedVotAttained = hasOperationalProfileVc(votToCheck, vcs);
+            }
+
+            if (requestedVotAttained) {
+                return Optional.of(votToCheck);
+            }
+        }
+        return Optional.empty();
+    }
+
+    public List<String> getIdentityConstituent(List<VerifiableCredential> tacticalVcs) {
+        return tacticalVcs.stream()
+                .map(
+                        vc -> {
+                            Cri cri = vc.getCri();
+                            if (cri.equals(Cri.DCMAW) || cri.equals(Cri.F2F)) {
+                                var credentialSubject = vc.getCredential().getCredentialSubject();
+                                if (credentialSubject instanceof PersonWithDocuments person) {
+                                    if (isNotEmpty(person.getBankAccount())) {
+                                        return cri.getId() + "-bankAccount";
+                                    } else if (isNotEmpty(person.getDrivingPermit())) {
+                                        return cri.getId() + "-drivingPermit";
+                                    } else if (isNotEmpty(person.getIdCard())) {
+                                        return cri.getId() + "-idCard";
+                                    } else if (isNotEmpty(person.getPassport())) {
+                                        return cri.getId() + "-passport";
+                                    } else if (isNotEmpty(person.getResidencePermit())) {
+                                        return cri.getId() + "-residencePermit";
+                                    } else if (isNotEmpty(person.getSocialSecurityRecord())) {
+                                        return cri.getId() + "-socialSecurityRecord";
+                                    }
+                                }
+                            }
+                            return cri.getId();
+                        })
+                .toList();
```

Comment:
It might be worth sorting these, unless you're confident that they will already be? i.e. so we don't end up mixing up `dcmaw,address,fraud` and `address,dcmaw,fraud`.

Reply:
Done